### DISCUSSION TITLE
Firebase notification arrives when app is open, but in idle state

### DISCUSSION THREAD

**Discussion by shubham-codestore** [invertase/react-native-firebase#6225] [Q&A]
Hi all,
I am new to react-native, and I am facing this weird problem when I implemented react-native firebase using below libraries - 
"@react-native-firebase/app": "^12.8.0"
"@react-native-firebase/messaging": "^12.8.0"

The problem is - 
When I am running my backend locally, the notification arrives immediately, doesn't matter on which screen I am.
The problem comes when I deployed my backend on AWS Lambda. Now, when I fire the notification, the app won't receive it until I interact with the app.

Please help me with this problem.

Thank You.

**Comment by mikehardy** [comment]
Hi there! These are old versions, so, careful with that.
App background states are really finicky. The phone vendors do everything they can to put your app to sleep to save battery, all the time.
You haven't even specified your test conditions though (type of device, versions etc) or code, or what exactly "idle state" means (is it, background but running? force closed? foreground but not tapping it?) no idea so can't help

New to react-native or not, software's software and the processes are the same, follow the scientific method and conduct real experiments with methods and materials as fixed as possible, and share real results

https://stackoverflow.com/help/how-to-ask
https://stackoverflow.com/help/minimal-reproducible-example
https://hackernoon.com/i-thought-i-understood-open-source-i-was-wrong-cf54999c097b

**Reply by shubham-codestore** [reply]
Sorry for the insufficient data.

I am facing the problem in both ios ( Iphone 7, version 14.6 ) and android ( Version 9 ).
Idle state means, when app is open, but not being interacted with.

Thank You.

**Reply by mikehardy** [reply]
Next step: Write a small bash script that uses the FCM REST API directly in order to send messages to your specific device tokens - something like the script here https://github.com/invertase/react-native-firebase/issues/4322

After removing the device keys please post a minimal App.js that reproduces the problem as well as the JSON that you send.

I am *guessing* that there is some subtle interaction somewhere and that your JSON is maybe missing priority settings or similar, but without specifying all the materials for an experiment a fellow scientist may only reasonably say "I don't know"
